libs/components/indicators/testing/src/wait/wait-harness.spec.ts

```diff
@@ -0,0 +1,133 @@
+        data-sky-id="wait-2"
+      ></sky-wait>
+    </div>
+  `,
+})
+class TestComponent {
+  public ariaLabel: string | undefined;
+
+  public isFullPage: boolean | undefined;
+
+  public isNonBlocking: boolean | undefined;
+
+  public isWaiting = false;
+}
+//#endregion Test component
+
+async function validateWaitProperties(
+  waitHarness: SkyWaitHarness,
+  fixture: ComponentFixture<TestComponent>,
+  isFullPage: boolean,
+  isNonBlocking: boolean,
+  ariaLabel?: string
+): Promise<void> {
+  fixture.componentInstance.ariaLabel = undefined;
+
+  fixture.componentInstance.isFullPage = isFullPage;
+  fixture.componentInstance.isNonBlocking = isNonBlocking;
+
+  if (ariaLabel) {
+    fixture.componentInstance.ariaLabel = ariaLabel;
+  }
+
+  fixture.componentInstance.isWaiting = true;
+  fixture.detectChanges();
+
+  const label =
+    fixture.componentInstance.ariaLabel ??
+    `${fixture.componentInstance.isFullPage ? 'Page l' : 'L'}oading.${
+      fixture.componentInstance.isNonBlocking ? '' : ' Please wait.'
+    }`;
+  await expectAsync(waitHarness.getAriaLabel()).toBeResolvedTo(label);
+  await expectAsync(waitHarness.isWaiting()).toBeResolvedTo(true);
+  await expectAsync(waitHarness.isFullPage()).toBeResolvedTo(isFullPage);
+  await expectAsync(waitHarness.isNonBlocking()).toBeResolvedTo(isNonBlocking);
+  fixture.componentInstance.isWaiting = false;
+}
+
+describe('Wait harness', () => {
+  async function setupTest(
+    options: { dataSkyId?: string; ariaLabel?: string } = {}
+  ): Promise<{
+    waitHarness: SkyWaitHarness;
+    fixture: ComponentFixture<TestComponent>;
+    loader: HarnessLoader;
+  }> {
+    await TestBed.configureTestingModule({
+      declarations: [TestComponent],
+      imports: [SkyWaitModule],
+    }).compileComponents();
+
+    const fixture = TestBed.createComponent(TestComponent);
+    const loader = TestbedHarnessEnvironment.loader(fixture);
+
+    let waitHarness: SkyWaitHarness;
+
+    if (options.dataSkyId) {
+      waitHarness = await loader.getHarness(
+        SkyWaitHarness.with({
+          dataSkyId: options.dataSkyId,
+        })
+      );
+    } else {
+      waitHarness = await loader.getHarness(SkyWaitHarness);
+    }
+
+    if (options.ariaLabel) {
+      fixture.componentInstance.ariaLabel = options.ariaLabel;
+    }
+
+    return { waitHarness, fixture, loader };
+  }
+
+  it('should return the expected ARIA label', async () => {
+    const { waitHarness, fixture } = await setupTest();
+    await validateWaitProperties(waitHarness, fixture, false, false);
+    await validateWaitProperties(waitHarness, fixture, true, false);
+    await validateWaitProperties(waitHarness, fixture, false, true);
+    await validateWaitProperties(waitHarness, fixture, true, true);
+    await validateWaitProperties(
+      waitHarness,
+      fixture,
+      true,
+      true,
+      'test label'
+    );
+  });
+
+  it('should get a wait by its data-sky-id property', async () => {
```

Comment:
I'm not sure if this description matches the test expectation.

Reply:
Split into 2 tests: one to validate the filter, one to validate the error